server/http/controller.go

```diff
@@ -277,6 +277,32 @@ func (c *controller) RunTest(ctx context.Context, testID string, runInformation
 	return openapi.Response(200, c.mappers.Out.Run(&run)), nil
 }
 
+func (c *controller) SetTestOutputs(ctx context.Context, testID string, testOutputs []openapi.TestOutput) (openapi.ImplResponse, error) {
+	test, err := c.testDB.GetLatestTestVersion(ctx, id.ID(testID))
+	if err != nil {
+		return handleDBError(err), err
+	}
+
+	newOutputs, err := c.mappers.In.Outputs(testOutputs)
+	if err != nil {
+		return openapi.Response(http.StatusBadRequest, err.Error()), nil
+	}
+
+	newTest, err := model.BumpVersionIfOutputsChanged(test, newOutputs)
```

Comment:
Question, do we need to keep sending the full body when updating a test? keeping track of the outputs and such? if so, I think instead of having two separate services to keep track of the same data we should stick with only one

Reply:
The `test update endpoint` updates everything. So, if we want to update only outputs we will need to send the trigger info, specs, etc. My idea here was to follow the same approach we have today for specs. We have a separate `PUT /definition endpoint` to set specs for a particular test.

Reply:
Makes sense!